src/c++/perf_analyzer/load_manager.cc

```diff
@@ -726,6 +744,17 @@ LoadManager::SetInputs(
         }
       }
     }
+
+    // If all optional inputs had data provided, this is a valid input. But if
+    // some inferences in the batch provided data for an optional input and some
+    // inferences did not, this is an invalid case and an error is thrown.
+    if (missing_data_cnt == 0) {
+      valid_inputs.push_back(input);
+    } else if (missing_data_cnt > 0 && missing_data_cnt < batch_size_) {
+      return cb::Error(
+          "Batch contained partial data for optional inputs. Batches sizes "
+          "larger than 1 must either provide all optional inputs or none.");
```

Comment:
Can you change the error message to:

"For batch sizes larger than 1, the same set of inputs must be specified for each batch. You cannot use different set of optional inputs for each individual batch."

Reply:
Changed 👍